kubernetes/deployment/controller_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright © 2018 Genome Research Limited
+// Author: Theo Barber-Bany
+// <tb15@sanger.ac.uk>.
+//
+//  This file is part of wr.
+//
+//  wr is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  wr is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with wr. If not, see <http://www.gnu.org/licenses/>.
+
+package deployment_test
+
+import (
+	"encoding/gob"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/VertebrateResequencing/wr/cloud"
+	"github.com/VertebrateResequencing/wr/internal"
+	"github.com/VertebrateResequencing/wr/jobqueue"
+	"github.com/VertebrateResequencing/wr/kubernetes/client"
+	kubedeployment "github.com/VertebrateResequencing/wr/kubernetes/deployment"
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/inconshreveable/log15"
+)
+
+var dc kubedeployment.Controller
+var autherr error
+var testingNamespace string
+
+// Just test that the call to Deploy() works, and that when configured as
+// expected, the deployment controller will copy the tarball, and that the
+// manager can be connected to.
+
+func init() {
+
+	dc = kubedeployment.Controller{
+		Client: &client.Kubernetesp{},
+	}
+	dc.Clientset, dc.Restconfig, autherr = dc.Client.Authenticate()
+	if autherr != nil {
+		panic(autherr)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	testingNamespace = strings.Replace(namesgenerator.GetRandomName(1), "_", "-", -1) + "-wr-testing"
+
+	_ = dc.Client.CreateNewNamespace(testingNamespace)
+
+	autherr = dc.Client.Initialize(dc.Clientset, testingNamespace)
+	if autherr != nil {
+		panic(autherr)
+	}
+
+}
+
+func TestDeploy(t *testing.T) {
+	cases := []struct {
+		containerImage  string
+		tempMountPath   string
+		cmdArgs         []string
+		configMountPath string
+		configMapData   string
+		requiredPorts   []int
+		resourcePath    string
+	}{
+		{
+			containerImage:  "ubuntu:latest",
+			tempMountPath:   "/wr-tmp/",
+			cmdArgs:         []string{"/wr-tmp/wr", "manager", "start", "-f"},
+			configMountPath: "/scripts/",
+			configMapData:   "echo \"hello world\"",
+			requiredPorts:   []int{8080, 8081},
+			resourcePath:    "/tmp/resources",
+		},
+	}
+	for _, c := range cases {
```

Comment:
Isn't this test function and its init() similar (identical?) to that in another test file? Can't you add this test case to that file or vice versa?

Reply:
I think I understand what you mean (identical to the init in the scheduler controller test), this way I try to get code coverage for both packages, is there a way to do it once but to cover both??